pkg/workspace/resource/node.go

```diff
@@ -127,23 +140,83 @@ func (c *NodeManager) getReadyNodesFromNodeClaims(ctx context.Context, wObj *kai
+			readyCount++
+		}
+	}
+
+	defer func() {
+		// Update the list of nodes, but don't block setting the condition on it.
+		if err := c.UpdateWorkerNodesInStatus(ctx, wObj, nodeList.Items); err != nil {
+			retErr = err
+		}
+	}()
+
+	if readyCount >= targetNodeCount {
+		// Enough Nodes are ready - update status condition to indicate success
+		if updateErr := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, kaitov1beta1.ConditionTypeNodeStatus, metav1.ConditionTrue,
+			"NodesReady", fmt.Sprintf("Enough Nodes are ready (TargetNodes: %d, CurrentReadyNodes: %d, SelectedNodes: %d)", targetNodeCount, readyCount, len(nodeList.Items))); updateErr != nil {
+			klog.ErrorS(updateErr, "failed to update Node status condition NodesReady to true", "workspace", klog.KObj(wObj))
+			return false, fmt.Errorf("failed to update Node status condition(NodesReady): %w", updateErr)
+		}
+		return true, nil
+	} else {
+		klog.InfoS("Ready Nodes for workspace are not enough currently", "workspace", client.ObjectKeyFromObject(wObj).String(),
+			"targetNodes", targetNodeCount, "currentReadyNodes", readyCount)
+		if updateErr := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, kaitov1beta1.ConditionTypeNodeStatus, metav1.ConditionFalse,
+			"NodeNotReady", fmt.Sprintf("Ready Nodes are not enough (TargetNodes: %d, CurrentReadyNodes: %d, SelectedNodes: %d)", targetNodeCount, readyCount, len(nodeList.Items))); updateErr != nil {
+			klog.ErrorS(updateErr, "failed to update Node status condition NodesReady to false", "workspace", klog.KObj(wObj))
+			return false, fmt.Errorf("failed to update Node status condition(NodeNotReady): %w", updateErr)
+		}
+
+		return false, nil
+	}
+}
+
+// propagateOwnedConditions checks the owned conditions, and if any one is false, it sets the condition to false with the reason and message of the first false condition found.
+func (c *NodeManager) propagateOwnedConditions(ctx context.Context, wObj *kaitov1beta1.Workspace, condition kaitov1beta1.ConditionType, conditionTypes []kaitov1beta1.ConditionType) error {
+	for _, cType := range conditionTypes {
+		ownedCondition := meta.FindStatusCondition(wObj.Status.Conditions, string(cType))
+		if ownedCondition == nil {
+			continue // Condition not found, skip to the next one
+		} else if ownedCondition.Status != metav1.ConditionTrue {
+			// Set the owned condition to false with the reason and message of the first false condition found.
+			if err := workspace.UpdateStatusConditionIfNotMatch(ctx, c.Client, wObj, condition, metav1.ConditionFalse, ownedCondition.Reason, ownedCondition.Message); err != nil {
+				klog.ErrorS(err, "failed to update workspace status", "workspace", klog.KObj(wObj))
+				return err
 			}
-			return err
+			return nil
 		}
 	}
 
+	return nil
+}
+
+// SetResourceReadyCondition updates the worker nodes list in workspace status.
+func (c *NodeManager) SetResourceReadyCondition(ctx context.Context, wObj *kaitov1beta1.Workspace) error {
```

Comment:
It is ok to refactor the code for this, but the name is confusing, maybe 

 SetResourceReadyCondition ->  SetResourceReadyConditionByStatus